### DISCUSSION TITLE
Infinite Scrolling

### DISCUSSION THREAD

**Discussion by Shubs21** [deso-protocol/dips#40] [Medium CIPs]
Infinite Scrolling could be useful for the following things.

1. Scrolling infinitely on the 'search by username bar'

2. Scrolling infinitely on the 'notifications panel'

I have attached images for context.

In image (1) I can't scroll beyond @jacksonxkrec.

In image (2) I can't scroll beyond @andreLoro's like notification.

<img width="359" alt="Screenshot 2021-07-11 at 10 17 31 PM" src="https://user-images.githubusercontent.com/87209627/125204353-76266200-e29a-11eb-8e91-8c80ea8fbc06.png">
<img width="601" alt="Screenshot 2021-07-11 at 10 17 54 PM" src="https://user-images.githubusercontent.com/87209627/125204356-79215280-e29a-11eb-82ba-4122ddef0d70.png">

**Comment by lazynina** [comment]
1. We currently only show the top 20 results ordered by clout. I'd be open to a PR that efficiently adds support for infinite scrolling on this components using the ngx-ui-scroll directive that has been adopted as the standard in the bitclout frontend project. This would also require some updates to paginate the [backend endpoint that supports search bar](https://github.com/bitclout/backend/blob/main/routes/user.go#L566)
2. Infinite scrolling is implemented on the notifications page.  if there's an issue with notification scrolling, please provide details on how to reproduce this issue including device type and any other relevant details.

**Reply by ItsAditya-xyz** [reply]
i can't seem to have infinite scrolling in notifications all the time (it only works sometimes). I am using brave browser (i don't think it's browser specific issue since the problem is same with chrome on mobile ). It's very rare that i can scroll down upto 50 notifications.

How to reproduce: if you can see infinite scrolling, reload the page. (i know it sounds weird, but it's what it is. Sometimes it works and mostly time it doesn't)

**Reply by lazynina** [reply]
I don't have this issue on chrome, safari, or brave on mobile, but I am using an iPhone. Even after reloading the page, I can still scroll down to the next page of notifications.

**Reply by tijno** [reply]
@lazynina On iOS for me it stops after 6 "loading" indicators (not counting the first) - using safari, latest (non beta) ios.

Ive tested this a few times.
